Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.h

```diff
@@ -0,0 +1,40 @@
+#ifndef FONTCONTROLSWIDGETSFORM_H
+#define FONTCONTROLSWIDGETSFORM_H
+
+#include <QWidget>
+
+namespace Ui {
+class FontControlsWidgetsForm;
+}
+
+class FontControlsWidgetsForm : public QWidget
+{
+    Q_OBJECT
+    Q_PROPERTY(QString text READ getText WRITE setText NOTIFY textChanged)
+    Q_PROPERTY(QString font READ getFont NOTIFY fontChanged)
+
+public:
+    explicit FontControlsWidgetsForm(QWidget *parent = nullptr);
+    ~FontControlsWidgetsForm();
+
+    QString getText();
+    void setText(const QString&);
+    QString getFont();
+
+signals:
+    void textChanged();
+    void fontChanged();
+    void on_pushButton_clicked();
+
+protected:
+    void closeEvent(QCloseEvent *event);
+
+private slots:
+    void on_lineEdit_textChanged(const QString &arg1);
```

Comment:
`on_*` is definitely not best practice (no error at compile time in case of a mismatch). Please use a real connect to PMF.

Reply:
Switched to explicit, modern, connects. Removed confusing `on_` prefix.
Are my declarations and casing correct or can these be improved?